packages/node-cli/src/core/sdksetup/SdkDownloadService.js

```diff
@@ -41,52 +45,73 @@ class SdkDownloadService {
 		const sdkDirectory = this._fileSystemService.createFolder(sdkParentDirectory, FOLDERS.NODE_CLI);
 
 		const fullURL = `${SdkProperties.getDownloadURL()}/${SdkProperties.getSdkFileName()}`;
+		const destinationFilePath = path.join(sdkDirectory, SdkProperties.getSdkFileName());
+		const proxy = process.env.SUITECLOUD_PROXY || process.env.npm_config_https_proxy || process.env.npm_config_proxy;
+		const skipProxy = SdkProperties.configFileExists();
 
 		try {
 			await executeWithSpinner({
-				action: this._downloadFile(fullURL, sdkDirectory),
+				action: this.downloadJarFilePromise(fullURL, destinationFilePath, proxy, skipProxy),
 				message: NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK, fullURL),
-			})
+			});
 			NodeConsoleLogger.info(NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK_SUCCESS));
-		}
-		catch (error) {
-			NodeConsoleLogger.error(NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK_ERROR, fullURL, unwrapExceptionMessage(error)))
+		} catch (error) {
+			NodeConsoleLogger.error(NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK_ERROR, fullURL, unwrapExceptionMessage(error)));
 		}
 	}
 
-	_downloadFile(url, sdkDirectory) {
-		const proxy = process.env.npm_config_https_proxy || process.env.npm_config_proxy;
-		const isProxyRequired = proxy && !SdkProperties.configFileExists();
-		const removeJarFilesFrom = this._removeJarFilesFrom;
+	downloadJarFilePromise(downloadUrl, destinationFilePath, proxy, skipProxy) {
+		const isProxyRequired = proxy && !skipProxy;
+		const downloadUrlObject = new URL(downloadUrl);
+		const downloadUrlProtocol = downloadUrlObject.protocol;
 
-		const options = {
-			method: 'GET',
-			uri: url,
+		const requestOptions = {
 			encoding: 'binary',
-			resolveWithFullResponse: true,
-			...(isProxyRequired && { proxy: proxy }),
+			...(isProxyRequired && { agent: new ProxyAgent(proxy, { tunnel: true, timeout: 15000 }) }),
 		};
 
-		return request(options).then(function(response) {
-			if (!VALID_JAR_CONTENT_TYPES.includes(response.headers['content-type'])) {
-				throw NodeTranslationService.getMessage(DOWNLOADING_SUITECLOUD_SDK_ERROR_FILE_NOT_AVAILABLE);
-			}
-
-			// remove all jar files before writing response to file
-			removeJarFilesFrom(sdkDirectory)
+		if (!/^http:$|^https:$/.test(downloadUrlProtocol)) {
+			throw new Error('Invalid SDK jar file download url protocol. Only http: or https: are allowed.');
+		}
 
-			const sdkDestinationFile = path.join(sdkDirectory, SdkProperties.getSdkFileName());
-			const file = fs.createWriteStream(sdkDestinationFile);
-			file.write(response.body, 'binary');
-			file.end();
+		const httpx = 'http:'.match(downloadUrlObject.protocol) ? http : https;
```

Comment:
What does it mean httpx? that it could be http or https?  could we named instead request?

Reply:
Similar to Antonio's question/answer, this is not yet a client/server or a the request. It's just the appropriate "library" that could be `http` or `https`.
But clearly we'll need to find a better naming or strategy, because both of you have doubts about it 🤣